src/lib/utils/socket/uri.h

```diff
@@ -0,0 +1,49 @@
+/*
+* (C) 2019 Nuno Goncalves <nunojpg@gmail.com>
+*
+* Botan is released under the Simplified BSD License (see license.txt)
+*/
+
+#ifndef BOTAN_URI_H_
+#define BOTAN_URI_H_
+
+#include <cstdint>
+#include <string>
+
+#include <botan/build.h>
+
+namespace Botan {
+
+struct BOTAN_TEST_API URI
```

Comment:
This class needs tests + a fuzzer

Reply:
I've changed this to use inet_pton. Is the fuzzer to test for memory leaks or crashes? Can you give me some advice about it?